symbolic-debuginfo/src/pe.rs

```diff
@@ -318,14 +309,33 @@ impl<'data> PeObject<'data> {
 
                 // See data specification:
                 // https://github.com/dotnet/runtime/blob/97ddb55e3adde20ceac579d935cef83cfe996169/docs/design/specs/PE-COFF.md#embedded-portable-pdb-debug-directory-entry-type-17
-                offset += 4; // skip `signature`
+                // let signature = &self.data[offset..(offset + 4)];
+                let mut signature: [u8; 4] = [0; 4];
+                self.data
+                    .gread_inout(&mut offset, &mut signature)
+                    .map_err(PeError::new)?;
+                if signature != "MPDB".as_bytes() {
+                    return Err(PeError::new(symbolic_ppdb::FormatError::from(
+                        symbolic_ppdb::FormatErrorKind::InvalidSignature,
+                    )));
+                }
                 let uncompressed_size: u32 = self
                     .data
                     .gread_with(&mut offset, LE)
                     .map_err(PeError::new)?;
-                let compressed_size = idd.size_of_data - 8; // 8 = the number bytes we have just read.
+
+                // 8 == the number bytes we have just read.
+                let compressed_size = idd.size_of_data as usize - 8;
```

Comment:
Actually, this can underflow if `size_of_data` (which is untrusted input) is <8.
It then overflows in `offset + compressed_size`, causing a panic because end < start.

```suggestion
                let Some(compressed_size) = (idd.size_of_data as usize).checked_sub(8) else { ... };
```

I know I’m very pedantic about this, just want to be sure we are as safe as possible. Also reminds me that we should maybe try to fuzz all our portable pdb work at some point just in case.
I have fuzzed the rest of symbolic quite some time ago, fixing some of these issues.

Reply:
added an explicit check above